tests/org.eclipse.e4.ui.tests/src/org/eclipse/e4/ui/tests/workbench/PartRenderingEngineTests.java

```diff
@@ -153,8 +153,11 @@ private void handle(Throwable ex, IEventLoopAdvisor advisor) {
 		try {
 			advisor.eventLoopException(ex);
 		} catch (Throwable t) {
-			if (t instanceof ThreadDeath td) {
-				throw td;
+			// The type ThreadDeath has been deprecated since version 20
+			// and marked for removal
+			if ("ThreadDeath".equals(t.getClass().getSimpleName())) {
```

Comment:
Is there a reason, why you didn't use the fully qualified name `java.lang.ThreadDeath`?
Using that would be saver.